code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/service/impl/ServiceDiscoveryServiceImpl.java

```diff
@@ -336,11 +336,23 @@ private String getInstanceName(Instance instance) {
     private Map<String, Object> getServiceDataAsMap(Service service) {
         Map<String, Object> originalData = new HashMap<>();
         originalData.putAll(DataUtils.getFields(service));
+
         Map<String, Object> data = new HashMap<>();
+
         Object launchConfig = originalData
                 .get(ServiceDiscoveryConstants.FIELD_LAUNCH_CONFIG);
         if (launchConfig != null) {
-            data.putAll((Map<? extends String, ? extends Object>) launchConfig);
+            Map<String, Object> configMap = (Map<String, Object>)launchConfig;
```

Comment:
Before, the putAll() transferred a Map which later gets the deployment unit labels added to it.  The side effect is that those labels makes its way into the service.  And when the 2nd deployment unit is processed, and the labels for that is merged with the service, we get a merging of the affinity labels.  This caused all the units to get deployed to the same host.  This was noticeable in the DB table.

We now provide a copy of the map to avoid inadvertent changes.